packages/youtube_player_iframe_web/lib/src/web_youtube_player_iframe_controller.dart

```diff
@@ -217,6 +216,15 @@ class YoutubePlayerIframeWeb extends PlatformWebViewWidget {
   }
 }
 
+String handleJsMessage(Object? jsMessage) {
+  return switch (jsMessage) {
+    String message => message,
+    Map map => jsonEncode(map),
```

Comment:
if it's already a map, do we need to encode and decode ?

Reply:
This is not a map everywhere... I've done this method to make sure it's gonna work with as less changes as possible in the existing logic

Reply:
I've done a separated handlers in this commit: [Improve web controller's JS message deserialization](https://github.com/sarbagyastha/youtube_player_flutter/pull/975/commits/280d9e7b49fa4a06b9d58daa349f75f7f70801d0)